### ISSUE TITLE
RKE1 Amazon EC2 clusters hang in provisioning with default node template ssh user

### ISSUE THREAD

**Issue by annablender** [rancher/rancher#40560]
**Setup**
- Rancher version: 2.6.3, 2.7.2
- Browser type & version: Safari

**Describe the bug**

An RKE1 Amazon EC2 cluster with a node template with the default SSH user (which is `ubuntu`) hangs in provisioning trying to reach the nodes.

Default is ubuntu
<img width="1123" alt="image" src="https://user-images.githubusercontent.com/12955547/218785613-cf51e292-b1b2-45d7-9a1e-4f50ab9e01f4.png">

On 2.6.3, the Rancher AMI docs https://github.com/rancher/os/blob/master/README.md/#user-content-amazon specify the default should be `rancher.`

If you don't change the SSH user in the node template, provisioning hangs at SSH connection
<img width="1543" alt="image" src="https://user-images.githubusercontent.com/12955547/218787004-777627f1-70a6-4c03-98fe-284750c6ad2b.png">

I tried changing nothing else in the node template and using the SSH user `rancher` instead and it works
<img width="1524" alt="image" src="https://user-images.githubusercontent.com/12955547/218790987-24b0e9ee-eeb5-4509-81be-8eea87caa74a.png">

**To Reproduce**

* Create an rke1 node template with default SSH user ubuntu
* Create rke1 cluster
* Watch `Machine Pools` tab

**Result**

RKE1 cluster hangs trying to connect to the nodes via SSH

**Expected Result**

RKE1 cluster should provision successfully.

**Notes**

I know this is manually configurable but it's not obvious why the rke1 cluster is hanging and it'll be easier for users if the default SSH user is set to `rancher.` Why is the default ubuntu in the first place? Was this done for another platform? Open to discussion. If this is updated, the fix will need to go into both 2.6.x and 2.7.x because I saw the issue on both versions of the ui.

**Comment by gaktive** [comment]
Transferred. 

@annablender I'm not sure if this is your team or not; if not, then @Jono-SUSE-Rancher which team needs to look at this?

**Comment by Jono-SUSE-Rancher** [comment]
I haven't done a super deep dive, but it looks like it would be team 2, yes.

**Comment by annablender** [comment]
Ahh ok, yes I'll transfer this issue and assign it to myself. Thank you!

**Comment by Oats87** [comment]
Why are we even referencing Rancher OS anymore? We should not be referencing Rancher OS, it is dead.

If a user wants to go out of their way to specify Rancher OS, then they should also see that they need to change the SSH user from `ubuntu` to `rancher` as in that documentation.

**Comment by Oats87** [comment]
If you are finding current references in the Rancher documentation to Rancher OS, we should remove that.

**Comment by annablender** [comment]
Thanks for the feedback @Oats87 

After looking into this a bit further, I believe it is not a regression. Rancher is set to use the ubuntu user by default and provision with no user configuration of the node template past security group and VPC. The `defaultSshUser` for amazonec2 config is set in rancher machine https://github.com/rancher/machine/blob/3961ee2c04211846a9c81b8e4b272ec4b86e73cb/drivers/amazonec2/amazonec2.go#L44. The RancherOS AMI list I pulled an AMI from in 2.6.3 is specifically for RancherOS and will only work with the sshUser `rancher.`

I tested this again on 2.6.3 and if you don't configure any AMI then Rancher will use a default ubuntu AMI that works with the default SSH user.

We could update the RancherOS AMI docs to discourage users from using it. Otherwise, this issue can be closed.
